src/util/dtos/ReadyGuildDTO.ts

```diff
@@ -35,6 +37,16 @@ export type ReadyUserGuildSettingsEntries = Omit<
 	channel_overrides: (ChannelOverride & { channel_id: string })[];
 };
 
+// TODO: probably should move somewhere else
+export interface ReadyPrivateChannel {
+	id: string;
+	flags: number;
+	is_spam: boolean;
+	last_message_id?: string;
+	recipients: PublicUser[];
+	type: ChannelType.DM | ChannelType.GROUP_DM;
```

Comment:
```suggestion
	type: ChannelType.DM | ChannelType.GROUP_DM | ChannelType.ENCRYPTED | ChannelType.ENCRYPTED_THREAD;
```

Reply:
Why is encrypted a channel type in the first place? Why not just use a flag on the channel itself, so any channel type can be encrypted?

Reply:
It is a different channel type because the server will need to accept certain event types regarding group key negotiation, and will also affect permissions. Also that E2EE requires a slightly different payload format (content in the attachment instead of content as a JSON field). All of which mean additional server-side code is required.

Reply:
All of that can be done using a flag on the channel rather than channel types imo. Using encrypted variants of channel types means
* We have to make sure our conditionals check for non-e2ee and e2ee variants of the same channel type, which would be messier
* Prevents showing more reasonable content in Discord.com clients. An encrypted channel could be displayed in the channel list correctly, but all content in the channel could be shown as "This is an encrypted channel blah blah". Using a separate channel type would just crash Discord.com clients
* For every new non-e2ee channel type, an e2ee variant must exist (annoying)
* I'm sure theres a fourth bullet point somewhere

Using an `encrypted` flag on the channel would address all of these

Reply:
@MaddyUnderStars An `encrypted` flag instead of a separate channel type would mean the server admin could forcibly downgrade an E2EE chat into a cleartext one.

Reply:
And no, no separate encrpyted variant for each channel type; just two types: `ENCRYPTED` for top level channels, and `ENCRYPTED_THREAD` for those with parent channels